SU-SWS/stanford_profile_helper
# D8CORE-2766: adding constraint for global message validation

# READY FOR REVIEW

# Summary
- Re-roll of https://github.com/SU-SWS/stanford_profile_helper/pull/73
- We change the simple form validator into an entity validator
- unit test included.

# Needed By (Date)
- 4/7

# Urgency
- Not critical.

# Steps to Test

1. Checkout this branch
2. clear your caches
3. Go to the global message screen, and turn on a global message, but don't fill out any fields.  Try to save. Notice an error.
4. Fill in at least one of the required fields.  Notice no error.

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
